src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -78535,6 +78535,61 @@ static class Extensions
             comp.VerifyDiagnostics();
         }
 
+        [Fact]
+        public void ForEach_ExtensionGetEnumerator16()
+        {
+            var source = @"
+using System.Collections.Generic;
+
+#nullable enable
+public class C<T> {
+    static void M(C<string> c)
+    {
+        foreach (var i in c)
+        {
+        }
+    }
+}
+#nullable disable
+public static class CExt
+{
+    public static IEnumerator<int> GetEnumerator<T>(this C<T> c, T t = default) => throw null;
+}";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                // (8,27): warning CS8620: Argument of type 'C<string>' cannot be used for parameter 'c' of type 'C<string?>' in 'IEnumerator<int> CExt.GetEnumerator<string?>(C<string?> c, string? t = null)' due to differences in the nullability of reference types.
+                //         foreach (var i in c)
+                Diagnostic(ErrorCode.WRN_NullabilityMismatchInArgument, "c").WithArguments("C<string>", "C<string?>", "c", "IEnumerator<int> CExt.GetEnumerator<string?>(C<string?> c, string? t = null)").WithLocation(8, 27)
```

Comment:
This is a new warning. Previously, we were not analyzing the default parameter. This matches the behavior you'd get if you did `c.GetEnumerator()` yourself.